src/mkpsxiso/iso.cpp

```diff
@@ -55,229 +56,26 @@ static cd::ISO_DATESTAMP GetISODateStamp(time_t time, signed char GMToffs)
 	return result;
 }
 
-int iso::DirTreeClass::GetWavSize(const std::filesystem::path& wavFile)
+int iso::DirTreeClass::GetAudioSize(const std::filesystem::path& audioFile)
 {
-	FILE *fp;
-
-	if ( !( fp = OpenFile( wavFile, "rb" ) ) )
-	{
-		printf("ERROR: File not found.\n");
-		return false;
-	}
-
-	// Get header chunk
-	struct
+	ma_decoder decoder;
+	VirtualWavEx vw;
+	bool isLossy;
+	if(ma_redbook_decoder_init_path_by_ext(audioFile, &decoder, &vw, isLossy) != MA_SUCCESS)
 	{
-		char	id[4];
-		int		size;
-		char	format[4];
-	} HeaderChunk;
-
-	fread( &HeaderChunk, 1, sizeof(HeaderChunk), fp );
-
-	if ( memcmp( &HeaderChunk.id, "RIFF", 4 ) ||
-		memcmp( &HeaderChunk.format, "WAVE", 4 ) )
-	{
-		// It must be a raw
-		fseek(fp, 0, SEEK_END);
-		int wavlen = ftell(fp);
-		fseek(fp, 0, SEEK_SET);
-
-		fclose( fp );
-		return wavlen;
-	}
-
-	// Get header chunk
-	struct
-	{
-		char	id[4];
-		int		size;
-		short	format;
-		short	chan;
-		int		freq;
-		int		brate;
-		short	balign;
-		short	bps;
-	} WAV_Subchunk1;
-
-	fread( &WAV_Subchunk1, 1, sizeof(WAV_Subchunk1), fp );
-
-	// Check if its a valid WAVE file
-	if ( memcmp( &WAV_Subchunk1.id, "fmt ", 4 ) )
-	{
-		fclose( fp );
 		return 0;
 	}
 
-	// Search for the data chunk
-	struct
+	const ma_uint64 expectedPCMFrames = ma_decoder_get_length_in_pcm_frames(&decoder);

```

Comment:
You can move `ma_decoder_uninit` right below this line, won't need to duplicate it twice below then.

Reply:
Done in G4Vi@71d5a63